tests/run/extmethods2.scala

```diff
@@ -0,0 +1,17 @@
+object Test extends App {
+
+  class TC
+
+  implied StringListOps given TC {
+    type T = List[String]
+    def (x: T) foo (y: T) = (x ++ y, the[TC])
```

Comment:
It's an interesting trick, but I'm afraid that it'll make signature information less readable in error messages and in the IDE (e.g. if I'm writing `a.foo(`, then a signature popup will tell me that an argument of type `T` is expected), and dealiasing all the time before showing type information is likely to be even less readable in general.

Reply:
Well, in practice it would probably be named `This` or `Self` or `Left`, not `T`. And of course we should have defined `foo` like this:
```scala
def (x: T) foo (y: List[String]) = ...
```
Then the signature popup gives you the right info again.